### ISSUE TITLE
load_extension weirdness in e_sqlite3

### ISSUE THREAD

**Issue by bricelam** [ericsink/SQLitePCL.raw#225]
``` sql
-- Works on Windows, fails on Linux/macOS
SELECT load_extension('mod_spatialite');

-- Works on Linux
SELECT load_extension('mod_spatialite.so');

-- Works on macOS
SELECT load_extension('mod_spatialite.dylib');
```

Can you think of anything special about the way e_sqlite3 is built that would preventing it from automatically adding the shared library extension on Linux and macOS?

The normal distributions of the sqlite3 command-line tool on these platforms work with just 'mod_spatialite'.

**Comment by ericsink** [comment]
Nothing I'm aware of would cause this.

I just checked the sqlite3.c code itself, and it looks like the behavior you want needs ___APPLE__ or the absence of define SQLITE_OS_UNIX.

Neither of these is mentioned in the compile time options page for sqlite:

https://www.sqlite.org/compile.html

So I don't mess with them.

But maybe I need to set these (or something similar) more explicitly.

**Comment by ericsink** [comment]
TLDR: I claim this is fixed in 1.1.12.

More detail:

1.  Using 1.1.11 (where e_sqlite3 is 3.22.0), I can repro this problem on Linux but not macOS.
2.  Using 1.1.12 (where e_sqlite3 is 3.26.0), the problem seems to be gone.
3.  Credit for the "fix", I believe, goes to SQLite itself.  The relevant section of code in sqlite3.c appears to have changed.

**Comment by bricelam** [comment]
It's fixed.
